File: src/Services/API/Domain/StoredProcedure/StoredProcedureMeta.cs
namespace API.Domain.StoredProcedure
{

    public class StoredProcedureMeta
    { 
        public IEnumerable<DependencyResult> Dependencies { get; set; } = new List<DependencyResult>();
        public IEnumerable<StoredProcedureParameter> Parameters { get; set; } = new List<StoredProcedureParameter>();
        public StoredProcedureCreateScript CreateScript { get; set; }
        public ExecutionPlanResult ExecutionPlan { get; set; }
    }

}

File: src/WebClient/mssql.client/src/app/procedure/model/StoredProcedureMeta.ts
import { DependencyResult } from './DependencyResult';
import { ExecutionPlanResult } from './ExecutionPlanResult';
import { StoredProcedureCreateScript } from './StoredProcedureCreateScript';
import { StoredProcedureParameter } from './StoredProcedureParameter';
import { StoredProcedureInfo } from './StoredProcedureInfo';
 
// Represents metadata for a stored procedure.
export class StoredProcedureMeta {
  dependencies: DependencyResult[] = [];
  parameters: StoredProcedureParameter[] = [];
  createScript: StoredProcedureCreateScript;
  executionPlan: ExecutionPlanResult;
  storedProcedureDependenciesTree: string = '';
  storedProcedureInfo: StoredProcedureInfo;
  constructor(init?: Partial<StoredProcedureMeta>) {
    Object.assign(this, init);
  }
}
